Performance

kalletlak opened the pull request:
1. Improve performance.
2. Code cleanup
   - Add dimension for each special chart.
   - Use dimension filtered data instead of looping to all data and applying filters to sync between charts.
   - Show loading image for special charts when filters are updated. 
   - Use dimension data to update special charts.

kalletlak committed:
Change logic in updating invisible charts

kalletlak committed:
Code Cleanup

1. Get rid of bridge chart, now just use the dimension.
2. Use invisible dimension to update special charts.
3. Update sync logic between groups.
4. Remove unwanted code.

kalletlak committed:
Fix pie chart qtip table loading

zhx828 commented on app/scripts/controller/sync.js:
```diff
@@ -71,6 +70,16 @@
+  iViz.sync.selectByCases = function(type_, data_, cases_) {
+    var caseIndices = iViz.getCaseIndices(type_);
+    var resultData_ = [];
+    $.each(cases_, function(key,val){
```
Can we use _.each for the performance and consistency?

kalletlak replied:
Sure! Done.

kalletlak committed:
Fixed issues

1. Fixed loading piechart qtip datatable
2. Fixed issue while clearing all filters
3. Replace #.each with _.each
4. Fixed issue while refreshing filtered scatter plot

kalletlak committed:
Small fixes

1. Rename invisibleChart to invisibleDimension in special charts.
2. Replace remaining $.each to _.each

kalletlak committed:
Changing sync logic between groups

kalletlak committed:
More changes

1. Replace invisible chart with just dimension.
2. Add dimension for table and scatter plots.
3. Update logic behind syncing chart groups.
4. Show loading image for special charts when filters are changed.
